mmv1/third_party/terraform/provider/provider_validators.go.tmpl

```diff
@@ -43,5 +46,44 @@ func ValidateEmptyStrings(v interface{}, k string) (warnings []string, errors []
 			fmt.Errorf("expected a non-empty string"))
 	}
 
+	return
+}
+
+func ValidateJWT(v interface{}, k string) (warnings []string, errors []error) {
```

Comment:
Were these validation functions taken from somewhere?

Without having acceptance tests specifically for auth, it would be great to have unit tests for these that we can expand upon

Reply:
these were added in to validate the JWT values, I've added a test in https://github.com/GoogleCloudPlatform/magic-modules/pull/12714/commits/bf764320a6104d07cd1b1aba026b27d03650a038 for this validation function.

Reply:
thanks!